charms/worker/k8s/src/config/extra_args.py

```diff
@@ -58,37 +70,34 @@ def craft(
             The configuration object to be updated with extra arguments.
         cluster_name (str): the name of the cluster to override in the extra arguments.
         node_ips (list[str]): the IP address of the node to override in the extra arguments.
+        datastore(Optional[str]): the name of the datastore.
     """
     if isinstance(dest, (BootstrapConfig, ControlPlaneNodeJoinConfig)):
-        cmd = _parse(src["kube-apiserver-extra-args"])
-        dest.extra_node_kube_apiserver_args = cmd
+        dest.extra_node_kube_apiserver_args = _parse(src["kube-apiserver-extra-args"])
 
-        cmd = _parse(src["kube-controller-manager-extra-args"])
+        args = _parse(src["kube-controller-manager-extra-args"])
         if cluster_name:
-            cmd.update(**{"--cluster-name": cluster_name})
+            args.update(**{"--cluster-name": cluster_name})
         else:
-            cmd.pop("--cluster-name", None)
-        dest.extra_node_kube_controller_manager_args = cmd
+            args.pop("--cluster-name", None)
+        dest.extra_node_kube_controller_manager_args = args
+
+        dest.extra_node_kube_scheduler_args = _parse(src["kube-scheduler-extra-args"])
 
-        cmd = _parse(src["kube-scheduler-extra-args"])
-        dest.extra_node_kube_scheduler_args = cmd
+        if datastore:
+            _configure_datastore_args(dest, src, datastore)
```

Comment:
what if you push the `if datastore:` into `_configure_datastore_args(...)`

it's already checking the value of datastore passed in...  if the value is `None`, i don't think `dest` will be updated (however src["datastore-extra-args"] would be read.